Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -610,3 +617,24 @@ void HexEditorWidget::drop_event(GUI::DropEvent& event)
         open_file(response.value().filename(), response.value().release_stream());
     }
 }
+
+ErrorOr<String> HexEditorWidget::make_display_string(DeprecatedString const& string)
+{
+    // FIXME: We probably want to replace all control characters here
+    StringBuilder builder;
+    for (auto c : string) {
```

Comment:
iteration type should probably depend on the use case, itʼd be good if one could provide encoding for the iterator? (thatʼs probably impossible as one i feel thereʼs only one codepoint iterator per string type)

it might be even that each string type will need a different `make_display_string`… this would make a case for [my other comment](https://github.com/SerenityOS/serenity/pull/17488#discussion_r1117828178)… 🤔